rs/ic_os/config/src/generate_testnet_config.rs

```diff
@@ -23,8 +23,8 @@ pub struct GenerateTestnetConfigArgs {
 
     // ICOSSettings arguments
     pub node_reward_type: Option<String>,
-    pub mgmt_mac: Option<String>,
-    pub deployment_environment: Option<String>,
+    pub mgmt_mac: Option<HwAddr>,
+    pub deployment_environment: Option<Deployment>,
```

Comment:
All the arguments in GenerateTestnetConfigArgs are strings so that the caller doesn't have to create any objects to call GenerateTestnetConfig, and can then be easily called from the command line. That's why we were parsing the mgmt_mac string into a FormattedMacAddress here.

Reply:
clap should be able to handle parsing these directly since they impl `FromStr`, in the same way that it is handling `Ipv6ConfigType`, `bool`, or `u64` too.

When we are in code, can we stick to using strong types? Any code users of `GenerateTestnetConfigArgs` should be able to use `.parse()?` in the same way.

Reply:
Yes, sounds good! It's one of my follow ups to get rid of GenerateTestnetConfig and instead have testing code construct a config struct more directly.